pydantic_settings/utils.py

```diff
@@ -0,0 +1,24 @@
+from pathlib import Path
+
+path_types = {
+    'is_dir': 'directory',
+    'is_file': 'file',
+    'is_mount': 'mount point',
+    'is_symlink': 'symlink',
+    'is_block_device': 'block device',
+    'is_char_device': 'char device',
+    'is_fifo': 'FIFO',
+    'is_socket': 'socket',
+}
+
+
+def path_type(p: Path) -> str:
```

Comment:
Minor thing, I'd suggest renaming this to `path_type_label` though, to indicate that the returned value is purely meant for display purposes, not for equality checks or anything like that.